caf-audit-service/src/main/java/com/hpe/caf/services/audit/server/api/impl/AuditeventsApiServiceImpl.java

```diff
@@ -0,0 +1,425 @@
+
+        //  Make sure the fixed audit event fields have been supplied.
+        if (isNullOrEmpty(newAuditEvent.getApplicationId())) {
+            LOG.error(ERR_MSG_APPLICATION_ID_NOT_SPECIFIED);
+            throw new BadRequestException(ERR_MSG_APPLICATION_ID_NOT_SPECIFIED);
+        }
+
+        if (isNullOrEmpty(newAuditEvent.getProcessId())) {
+            LOG.error(ERR_MSG_PROCESS_ID_NOT_SPECIFIED);
+            throw new BadRequestException(ERR_MSG_PROCESS_ID_NOT_SPECIFIED);
+        }
+
+        if (newAuditEvent.getThreadId() == null || 0 == newAuditEvent.getThreadId()) {
+            LOG.error(ERR_MSG_THREAD_ID_NOT_SPECIFIED);
+            throw new BadRequestException(ERR_MSG_THREAD_ID_NOT_SPECIFIED);
+        }
+
+        if (newAuditEvent.getEventOrder() == null || 0 == newAuditEvent.getEventOrder()) {
+            LOG.error(ERR_MSG_EVENT_ORDER_NOT_SPECIFIED);
+            throw new BadRequestException(ERR_MSG_EVENT_ORDER_NOT_SPECIFIED);
+        }
+
+        if (isNullOrEmpty(newAuditEvent.getEventTime())) {
+            LOG.error(ERR_MSG_EVENT_TIME_NOT_SPECIFIED);
+            throw new BadRequestException(ERR_MSG_EVENT_TIME_NOT_SPECIFIED);
+        }
+
+        if (isNullOrEmpty(newAuditEvent.getEventTimeSource())) {
+            LOG.error(ERR_MSG_EVENT_TIME_SOURCE_NOT_SPECIFIED);
+            throw new BadRequestException(ERR_MSG_EVENT_TIME_SOURCE_NOT_SPECIFIED);
+        }
+
+        if (isNullOrEmpty(newAuditEvent.getUserId())) {
+            LOG.error(ERR_MSG_USER_ID_NOT_SPECIFIED);
+            throw new BadRequestException(ERR_MSG_USER_ID_NOT_SPECIFIED);
+        }
+
+        if (isNullOrEmpty(newAuditEvent.getTenantId())) {
+            LOG.error(ERR_MSG_TENANT_ID_NOT_SPECIFIED);
+            throw new BadRequestException(ERR_MSG_TENANT_ID_NOT_SPECIFIED);
+        }
+
+        if (isNullOrEmpty(newAuditEvent.getCorrelationId())) {
+            LOG.error(ERR_MSG_CORRELATION_ID_NOT_SPECIFIED);
+            throw new BadRequestException(ERR_MSG_CORRELATION_ID_NOT_SPECIFIED);
+        }
+
+        if (isNullOrEmpty(newAuditEvent.getEventTypeId())) {
+            LOG.error(ERR_MSG_EVENT_TYPE_ID_NOT_SPECIFIED);
+            throw new BadRequestException(ERR_MSG_EVENT_TYPE_ID_NOT_SPECIFIED);
+        }
+
+        if (isNullOrEmpty(newAuditEvent.getEventCategoryId())) {
+            LOG.error(ERR_MSG_EVENT_CATEGORY_ID_NOT_SPECIFIED);
+            throw new BadRequestException(ERR_MSG_EVENT_CATEGORY_ID_NOT_SPECIFIED);
+        }
+
+        //  Make sure at least one custom audit event field has been supplied.
+        if (newAuditEvent.getEventParams().isEmpty()) {
+            LOG.error(ERR_MSG_CUSTOM_FIELDS_NOT_SPECIFIED);
+            throw new BadRequestException(ERR_MSG_CUSTOM_FIELDS_NOT_SPECIFIED);
+        }
+    }
+
+    /**
+     * Returns TRUE if the specified string is null or empty, otherwise FALSE.
+     */
+    private boolean isNullOrEmpty(String str) {
+        return str == null || str.isEmpty();
+    }
```

Comment:
No need to to change this at all as I am not sure if we are permitted to use Common Lang StringUtils in HPE.  But Common Lang StringUtils is very handy for all this kind of stuff:  https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html